devbox.go

```diff
@@ -16,41 +16,36 @@ import (
 
 // Devbox provides an isolated development environment.
 type Devbox interface {
-	// Add adds Nix packages to the config so that they're available in the devbox
-	// environment. It validates that the Nix packages exist, and install them.
-	// Adding duplicate packages is a no-op.
 	Add(ctx context.Context, platforms, excludePlatforms []string, pkgs ...string) error
 	Config() *devconfig.Config
-	ProjectDir() string
-	// Generate creates the directory of Nix files and the Dockerfile that define
-	// the devbox environment.
-	Generate(ctx context.Context) error
-	GenerateDevcontainer(ctx context.Context, generateOpts devopt.GenerateOpts) error
-	GenerateDockerfile(ctx context.Context, generateOpts devopt.GenerateOpts) error
-	GenerateEnvrcFile(ctx context.Context, force bool, envFlags devopt.EnvFlags) error
-	Info(ctx context.Context, pkg string, markdown bool) error
+	EnvVars(ctx context.Context) ([]string, error)
+	Info(ctx context.Context, pkg string, markdown bool) (string, error)
 	Install(ctx context.Context) error
 	IsEnvEnabled() bool
 	ListScripts() []string
-	PrintEnv(ctx context.Context, includeHooks bool) (string, error)
-	PrintEnvVars(ctx context.Context) ([]string, error)
-	PrintGlobalList() error
+	NixEnv(ctx context.Context, includeHooks bool) (string, error)
+	PackageNames() []string
```

Comment:
Are these config PackageNames or installable PackageNames?

Reply:
Just what is in config. This is a no-op (PrintGlobaList used to call PackageNames).

We can chat about whether this is correct or not (I think it's fine), but would rather not change it in this PR.